thamos/lib.py

```diff
@@ -183,15 +183,17 @@ def advise(
             "Cannot use runtime_environment and runtime_environment_name at the same time"
         )
 
+    if runtime_environment is None:
+        runtime_environment = thoth_config.get_runtime_environment(runtime_environment_name)
+
     # We use the explicit one if provided at the end.
     if limit_latest_versions is None:
         limit_latest_versions = thoth_config.content.get("limit_latest_versions")
 
     if recommendation_type is None:
-        recommendation_type = thoth_config.content.get("recommendation_type") or "stable"
-
-    if runtime_environment is None:
-        runtime_environment = thoth_config.get_runtime_environment(runtime_environment_name)
+        recommendation_type = thoth_config.content.get(
+            "recommendation_type"
+        ) or runtime_environment.pop("recommendation_type", "stable")
```

Comment:
I think pop'ing `recommendation_type` will need to be done always to preserve case when it stays in the `runtime_envrionment` dict (retrieved from config file and `recommendation_type` is set on this function call). So something like (after this if) should fix that:

```python
runtime_environment.pop("recommendation_type", None)
```

Reply:
Ah, I see, so how about reverting and on line 205:

```
if "recommendation_type" in runtime_environment and recommendation_type is None:
```
we just got rid of the `and` part of the condition -- the `recommendation_type` will never be `None`?

Reply:
Sounds like another approach :+1:

Reply:
Cool, thanks!